// @ariakit/react/cjs/__chunks/4N2JD2M7.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});

var _D4HQPWQGcjs = require('./D4HQPWQG.cjs');




var _5X2KE4NBcjs = require('./5X2KE4NB.cjs');


var _FWOU7QANcjs = require('./FWOU7QAN.cjs');


var _TJNZ4CWTcjs = require('./TJNZ4CWT.cjs');








var _7FULUV5Lcjs = require('./7FULUV5L.cjs');

// ../ariakit-react-components/dist/popover/popover-disclosure.js
var _jsxruntime = require('react/jsx-runtime');
var TagName = "button";
var usePopoverDisclosure = _7FULUV5Lcjs.createHook.call(void 0, function usePopoverDisclosure2({ store, ...props }) {
  const context = _5X2KE4NBcjs.usePopoverProviderContext.call(void 0, );
  store = store || context;
  _7FULUV5Lcjs.invariant.call(void 0, store, process.env.NODE_ENV !== "production" && "PopoverDisclosure must receive a `store` prop or be wrapped in a PopoverProvider component.");
  const onClickProp = props.onClick;
  const onClick = _7FULUV5Lcjs.useEvent.call(void 0, (event) => {
    store == null ? void 0 : store.setAnchorElement(event.currentTarget);
    onClickProp == null ? void 0 : onClickProp(event);
  });
  props = _7FULUV5Lcjs.useWrapElement.call(void 0, props, (element) => /* @__PURE__ */ _jsxruntime.jsx.call(void 0, _5X2KE4NBcjs.PopoverScopedContextProvider, {
    value: store,
    children: element
  }), [store]);
  props = {
    ...props,
    onClick
  };
  props = _D4HQPWQGcjs.usePopoverAnchor.call(void 0, {
    store,
    ...props
  });
  props = _FWOU7QANcjs.useDialogDisclosure.call(void 0, {
    store,
    ...props
  });
  return props;
});
var PopoverDisclosure = _7FULUV5Lcjs.forwardRef.call(void 0, function PopoverDisclosure2(props) {
  return _7FULUV5Lcjs.createElement.call(void 0, TagName, usePopoverDisclosure(props));
});

// ../ariakit-react-components/dist/popover/popover-disclosure-arrow.js
var _react = require('react');

var TagName2 = "span";
var pointsMap = {
  top: "4,10 8,6 12,10",
  right: "6,4 10,8 6,12",
  bottom: "4,6 8,10 12,6",
  left: "10,4 6,8 10,12"
};
var usePopoverDisclosureArrow = _7FULUV5Lcjs.createHook.call(void 0, function usePopoverDisclosureArrow2({ store, placement, ...props }) {
  const context = _5X2KE4NBcjs.usePopoverContext.call(void 0, );
  store = store || context;
  _7FULUV5Lcjs.invariant.call(void 0, store, process.env.NODE_ENV !== "production" && "PopoverDisclosureArrow must be wrapped in a PopoverDisclosure component.");
  const points = pointsMap[_TJNZ4CWTcjs.useStoreState.call(void 0, store, (state) => placement || state.placement).split("-")[0]];
  props = {
    children: _react.useMemo.call(void 0, () => /* @__PURE__ */ _jsxruntime.jsx.call(void 0, "svg", {
      display: "block",
      fill: "none",
      stroke: "currentColor",
      strokeLinecap: "round",
      strokeLinejoin: "round",
      strokeWidth: 1.5,
      viewBox: "0 0 16 16",
      height: "1em",
      width: "1em",
      children: /* @__PURE__ */ _jsxruntime.jsx.call(void 0, "polyline", { points })
    }), [points]),
    "aria-hidden": true,
    ...props,
    style: {
      width: "1em",
      height: "1em",
      pointerEvents: "none",
      ...props.style
    }
  };
  return _7FULUV5Lcjs.removeUndefinedValues.call(void 0, props);
});
var PopoverDisclosureArrow = _7FULUV5Lcjs.forwardRef.call(void 0, function PopoverDisclosureArrow2(props) {
  return _7FULUV5Lcjs.createElement.call(void 0, TagName2, usePopoverDisclosureArrow(props));
});






exports.usePopoverDisclosure = usePopoverDisclosure; exports.PopoverDisclosure = PopoverDisclosure; exports.usePopoverDisclosureArrow = usePopoverDisclosureArrow; exports.PopoverDisclosureArrow = PopoverDisclosureArrow;
