// @ariakit/react/cjs/__chunks/AUYZXX2M.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});

var _TQJOLFOScjs = require('./TQJOLFOS.cjs');














var _7FULUV5Lcjs = require('./7FULUV5L.cjs');

// ../ariakit-react-components/dist/portal/portal-context.js
var _react = require('react');
var PortalContext = _react.createContext.call(void 0, null);

// ../ariakit-react-components/dist/portal/portal.js

var _jsxruntime = require('react/jsx-runtime');
var _reactdom = require('react-dom');
var TagName = "div";
function getRootElement(element) {
  const doc = _7FULUV5Lcjs.getDocument.call(void 0, element);
  const { fullscreenElement } = doc;
  if (fullscreenElement instanceof HTMLElement) return fullscreenElement;
  return doc.body;
}
function getPortalElement(element, portalElement) {
  if (!portalElement) return _7FULUV5Lcjs.getDocument.call(void 0, element).createElement("div");
  if (typeof portalElement === "function") return portalElement(element);
  return portalElement;
}
function getRandomId(prefix = "id") {
  return `${prefix ? `${prefix}-` : ""}${Math.random().toString(36).slice(2, 8)}`;
}
function queueFocus(element) {
  queueMicrotask(() => {
    element == null ? void 0 : element.focus();
  });
}
var usePortal = _7FULUV5Lcjs.createHook.call(void 0, function usePortal2({ preserveTabOrder, preserveTabOrderAnchor, portalElement, portalRef, portal = true, ...props }) {
  const ref = _react.useRef.call(void 0, null);
  const refProp = _7FULUV5Lcjs.useMergeRefs.call(void 0, ref, props.ref);
  const context = _react.useContext.call(void 0, PortalContext);
  const [portalNode, setPortalNode] = _react.useState.call(void 0, null);
  const [anchorPortalNode, setAnchorPortalNode] = _react.useState.call(void 0, null);
  const outerBeforeRef = _react.useRef.call(void 0, null);
  const innerBeforeRef = _react.useRef.call(void 0, null);
  const innerAfterRef = _react.useRef.call(void 0, null);
  const outerAfterRef = _react.useRef.call(void 0, null);
  _7FULUV5Lcjs.useSafeLayoutEffect.call(void 0, () => {
    const element = ref.current;
    if (!element || !portal) {
      setPortalNode(null);
      return;
    }
    const portalEl = getPortalElement(element, portalElement);
    if (!portalEl) {
      setPortalNode(null);
      return;
    }
    const isPortalInDocument = portalEl.isConnected;
    if (!isPortalInDocument) (context || getRootElement(element)).appendChild(portalEl);
    if (!portalEl.id) portalEl.id = element.id ? `portal/${element.id}` : getRandomId();
    setPortalNode(portalEl);
    const cleanup = _7FULUV5Lcjs.setRef.call(void 0, portalRef, portalEl);
    return () => {
      if (typeof cleanup === "function") cleanup();
      else _7FULUV5Lcjs.setRef.call(void 0, portalRef, null);
      if (!isPortalInDocument) portalEl.remove();
    };
  }, [
    portal,
    portalElement,
    context,
    portalRef
  ]);
  _react.useEffect.call(void 0, () => {
    if (!portalNode) return;
    if (context) return;
    if (portalElement) return;
    const doc = _7FULUV5Lcjs.getDocument.call(void 0, portalNode);
    const onFullscreenChange = () => {
      const rootElement = getRootElement(portalNode);
      if (portalNode.parentElement !== rootElement) rootElement.appendChild(portalNode);
    };
    onFullscreenChange();
    doc.addEventListener("fullscreenchange", onFullscreenChange);
    return () => {
      doc.removeEventListener("fullscreenchange", onFullscreenChange);
    };
  }, [
    portalNode,
    context,
    portalElement
  ]);
  _7FULUV5Lcjs.useSafeLayoutEffect.call(void 0, () => {
    if (!portal) return;
    if (!preserveTabOrder) return;
    if (!preserveTabOrderAnchor) return;
    const element = _7FULUV5Lcjs.getDocument.call(void 0, preserveTabOrderAnchor).createElement("span");
    element.style.position = "fixed";
    preserveTabOrderAnchor.insertAdjacentElement("afterend", element);
    setAnchorPortalNode(element);
    return () => {
      element.remove();
      setAnchorPortalNode(null);
    };
  }, [
    portal,
    preserveTabOrder,
    preserveTabOrderAnchor
  ]);
  _react.useEffect.call(void 0, () => {
    if (!portalNode) return;
    if (!preserveTabOrder) return;
    let raf = 0;
    const onFocus = (event) => {
      if (!_7FULUV5Lcjs.isFocusEventOutside.call(void 0, event)) return;
      const focusing = event.type === "focusin";
      cancelAnimationFrame(raf);
      if (focusing) return _7FULUV5Lcjs.restoreFocusIn.call(void 0, portalNode);
      raf = requestAnimationFrame(() => {
        _7FULUV5Lcjs.disableFocusIn.call(void 0, portalNode, true);
      });
    };
    portalNode.addEventListener("focusin", onFocus, true);
    portalNode.addEventListener("focusout", onFocus, true);
    return () => {
      cancelAnimationFrame(raf);
      portalNode.removeEventListener("focusin", onFocus, true);
      portalNode.removeEventListener("focusout", onFocus, true);
    };
  }, [portalNode, preserveTabOrder]);
  props = _7FULUV5Lcjs.useWrapElement.call(void 0, props, (element) => {
    element = /* @__PURE__ */ _jsxruntime.jsx.call(void 0, PortalContext.Provider, {
      value: portalNode || context,
      children: element
    });
    if (!portal) return element;
    if (!portalNode) return /* @__PURE__ */ _jsxruntime.jsx.call(void 0, "span", {
      ref: refProp,
      id: props.id,
      style: { position: "fixed" },
      hidden: true
    });
    element = /* @__PURE__ */ _jsxruntime.jsxs.call(void 0, _jsxruntime.Fragment, { children: [
      preserveTabOrder && portalNode && /* @__PURE__ */ _jsxruntime.jsx.call(void 0, _TQJOLFOScjs.FocusTrap, {
        ref: innerBeforeRef,
        "data-focus-trap": props.id,
        className: "__focus-trap-inner-before",
        onFocus: (event) => {
          if (_7FULUV5Lcjs.isFocusEventOutside.call(void 0, event, portalNode)) queueFocus(_7FULUV5Lcjs.getNextTabbable.call(void 0, ));
          else queueFocus(outerBeforeRef.current);
        }
      }),
      element,
      preserveTabOrder && portalNode && /* @__PURE__ */ _jsxruntime.jsx.call(void 0, _TQJOLFOScjs.FocusTrap, {
        ref: innerAfterRef,
        "data-focus-trap": props.id,
        className: "__focus-trap-inner-after",
        onFocus: (event) => {
          if (_7FULUV5Lcjs.isFocusEventOutside.call(void 0, event, portalNode)) queueFocus(_7FULUV5Lcjs.getPreviousTabbable.call(void 0, ));
          else queueFocus(outerAfterRef.current);
        }
      })
    ] });
    if (portalNode) element = _reactdom.createPortal.call(void 0, element, portalNode);
    let preserveTabOrderElement = /* @__PURE__ */ _jsxruntime.jsxs.call(void 0, _jsxruntime.Fragment, { children: [
      preserveTabOrder && portalNode && /* @__PURE__ */ _jsxruntime.jsx.call(void 0, _TQJOLFOScjs.FocusTrap, {
        ref: outerBeforeRef,
        "data-focus-trap": props.id,
        className: "__focus-trap-outer-before",
        onFocus: (event) => {
          if (!(event.relatedTarget === outerAfterRef.current) && _7FULUV5Lcjs.isFocusEventOutside.call(void 0, event, portalNode)) queueFocus(innerBeforeRef.current);
          else queueFocus(_7FULUV5Lcjs.getPreviousTabbable.call(void 0, ));
        }
      }),
      preserveTabOrder && /* @__PURE__ */ _jsxruntime.jsx.call(void 0, "span", {
        "aria-owns": portalNode == null ? void 0 : portalNode.id,
        style: { position: "fixed" }
      }),
      preserveTabOrder && portalNode && /* @__PURE__ */ _jsxruntime.jsx.call(void 0, _TQJOLFOScjs.FocusTrap, {
        ref: outerAfterRef,
        "data-focus-trap": props.id,
        className: "__focus-trap-outer-after",
        onFocus: (event) => {
          if (_7FULUV5Lcjs.isFocusEventOutside.call(void 0, event, portalNode)) queueFocus(innerAfterRef.current);
          else {
            const nextTabbable = _7FULUV5Lcjs.getNextTabbable.call(void 0, );
            if (nextTabbable === innerBeforeRef.current) {
              requestAnimationFrame(() => {
                var _a;
                return (_a = _7FULUV5Lcjs.getNextTabbable.call(void 0, )) == null ? void 0 : _a.focus();
              });
              return;
            }
            queueFocus(nextTabbable);
          }
        }
      })
    ] });
    if (anchorPortalNode && preserveTabOrder) preserveTabOrderElement = _reactdom.createPortal.call(void 0, preserveTabOrderElement, anchorPortalNode);
    return /* @__PURE__ */ _jsxruntime.jsxs.call(void 0, _jsxruntime.Fragment, { children: [preserveTabOrderElement, element] });
  }, [
    portalNode,
    context,
    portal,
    props.id,
    preserveTabOrder,
    anchorPortalNode
  ]);
  props = {
    ...props,
    ref: refProp
  };
  return props;
});
var Portal = _7FULUV5Lcjs.forwardRef.call(void 0, function Portal2(props) {
  return _7FULUV5Lcjs.createElement.call(void 0, TagName, usePortal(props));
});





exports.PortalContext = PortalContext; exports.usePortal = usePortal; exports.Portal = Portal;
