// @ariakit/react/cjs/__chunks/CZX7RHRT.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});

var _5X2KE4NBcjs = require('./5X2KE4NB.cjs');



var _DKS5EE6Lcjs = require('./DKS5EE6L.cjs');




var _7FULUV5Lcjs = require('./7FULUV5L.cjs');

// ../ariakit-react-components/dist/popover/popover-dismiss.js
var TagName = "button";
var usePopoverDismiss = _7FULUV5Lcjs.createHook.call(void 0, function usePopoverDismiss2({ store, ...props }) {
  const context = _5X2KE4NBcjs.usePopoverScopedContext.call(void 0, );
  store = store || context;
  props = _DKS5EE6Lcjs.useDialogDismiss.call(void 0, {
    store,
    ...props
  });
  return props;
});
var PopoverDismiss = _7FULUV5Lcjs.forwardRef.call(void 0, function PopoverDismiss2(props) {
  return _7FULUV5Lcjs.createElement.call(void 0, TagName, usePopoverDismiss(props));
});

// ../ariakit-react-components/dist/popover/popover-heading.js
var TagName2 = "h1";
var usePopoverHeading = _7FULUV5Lcjs.createHook.call(void 0, function usePopoverHeading2(props) {
  props = _DKS5EE6Lcjs.useDialogHeading.call(void 0, props);
  return props;
});
var PopoverHeading = _7FULUV5Lcjs.forwardRef.call(void 0, function PopoverHeading2(props) {
  return _7FULUV5Lcjs.createElement.call(void 0, TagName2, usePopoverHeading(props));
});






exports.usePopoverDismiss = usePopoverDismiss; exports.PopoverDismiss = PopoverDismiss; exports.usePopoverHeading = usePopoverHeading; exports.PopoverHeading = PopoverHeading;
