// @ariakit/react/cjs/__chunks/GEG7MVG3.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});

var _MU4IYYTFcjs = require('./MU4IYYTF.cjs');


var _IMPJZ3WWcjs = require('./IMPJZ3WW.cjs');



var _CZX7RHRTcjs = require('./CZX7RHRT.cjs');




var _7FULUV5Lcjs = require('./7FULUV5L.cjs');

// ../ariakit-react-components/dist/hovercard/hovercard-description.js
var TagName = "p";
var useHovercardDescription = _7FULUV5Lcjs.createHook.call(void 0, function useHovercardDescription2(props) {
  props = _IMPJZ3WWcjs.usePopoverDescription.call(void 0, props);
  return props;
});
var HovercardDescription = _7FULUV5Lcjs.forwardRef.call(void 0, function HovercardDescription2(props) {
  return _7FULUV5Lcjs.createElement.call(void 0, TagName, useHovercardDescription(props));
});

// ../ariakit-react-components/dist/hovercard/hovercard-dismiss.js
var TagName2 = "button";
var useHovercardDismiss = _7FULUV5Lcjs.createHook.call(void 0, function useHovercardDismiss2({ store, ...props }) {
  const context = _MU4IYYTFcjs.useHovercardScopedContext.call(void 0, );
  store = store || context;
  props = _CZX7RHRTcjs.usePopoverDismiss.call(void 0, {
    store,
    ...props
  });
  return props;
});
var HovercardDismiss = _7FULUV5Lcjs.forwardRef.call(void 0, function HovercardDismiss2(props) {
  return _7FULUV5Lcjs.createElement.call(void 0, TagName2, useHovercardDismiss(props));
});

// ../ariakit-react-components/dist/hovercard/hovercard-heading.js
var TagName3 = "h1";
var useHovercardHeading = _7FULUV5Lcjs.createHook.call(void 0, function useHovercardHeading2(props) {
  props = _CZX7RHRTcjs.usePopoverHeading.call(void 0, props);
  return props;
});
var HovercardHeading = _7FULUV5Lcjs.forwardRef.call(void 0, function HovercardHeading2(props) {
  return _7FULUV5Lcjs.createElement.call(void 0, TagName3, useHovercardHeading(props));
});








exports.useHovercardDescription = useHovercardDescription; exports.HovercardDescription = HovercardDescription; exports.useHovercardDismiss = useHovercardDismiss; exports.HovercardDismiss = HovercardDismiss; exports.useHovercardHeading = useHovercardHeading; exports.HovercardHeading = HovercardHeading;
