// @ariakit/react/cjs/__chunks/HYE4EQBZ.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});




var _MU4IYYTFcjs = require('./MU4IYYTF.cjs');


var _XIMN7GOEcjs = require('./XIMN7GOE.cjs');


var _FWOU7QANcjs = require('./FWOU7QAN.cjs');


var _O72M6F4Zcjs = require('./O72M6F4Z.cjs');


var _TJNZ4CWTcjs = require('./TJNZ4CWT.cjs');











var _7FULUV5Lcjs = require('./7FULUV5L.cjs');

// ../ariakit-react-components/dist/hovercard/hovercard-arrow.js
var TagName = "div";
var useHovercardArrow = _7FULUV5Lcjs.createHook.call(void 0, function useHovercardArrow2({ store, ...props }) {
  const context = _MU4IYYTFcjs.useHovercardContext.call(void 0, );
  store = store || context;
  props = _XIMN7GOEcjs.usePopoverArrow.call(void 0, {
    store,
    ...props
  });
  return props;
});
var HovercardArrow = _7FULUV5Lcjs.forwardRef.call(void 0, function HovercardArrow2(props) {
  return _7FULUV5Lcjs.createElement.call(void 0, TagName, useHovercardArrow(props));
});

// ../ariakit-react-components/dist/hovercard/hovercard-disclosure.js
var _react = require('react');
var _jsxruntime = require('react/jsx-runtime');
var TagName2 = "button";
var useHovercardDisclosure = _7FULUV5Lcjs.createHook.call(void 0, function useHovercardDisclosure2({ store, ...props }) {
  const context = _MU4IYYTFcjs.useHovercardProviderContext.call(void 0, );
  store = store || context;
  _7FULUV5Lcjs.invariant.call(void 0, store, process.env.NODE_ENV !== "production" && "HovercardDisclosure must receive a `store` prop or be wrapped in a HovercardProvider component.");
  const [visible, setVisible] = _react.useState.call(void 0, false);
  _react.useEffect.call(void 0, () => {
    if (!visible) return;
    const onBlur = (event) => {
      if (!store) return;
      const nextActiveElement = event.relatedTarget;
      if (_7FULUV5Lcjs.isNode.call(void 0, nextActiveElement)) {
        const { anchorElement: anchor, popoverElement: popover, disclosureElement: disclosure } = store.getState();
        if (anchor && _7FULUV5Lcjs.contains.call(void 0, anchor, nextActiveElement)) return;
        if (popover && _7FULUV5Lcjs.contains.call(void 0, popover, nextActiveElement)) return;
        if (disclosure && _7FULUV5Lcjs.contains.call(void 0, disclosure, nextActiveElement)) return;
        if (_7FULUV5Lcjs.isElement.call(void 0, nextActiveElement) && nextActiveElement.hasAttribute("data-focus-trap")) return;
      }
      setVisible(false);
    };
    return _7FULUV5Lcjs.addGlobalEventListener.call(void 0, "focusout", onBlur, true);
  }, [visible, store]);
  _react.useEffect.call(void 0, () => {
    return _TJNZ4CWTcjs.sync.call(void 0, store, ["anchorElement"], (state) => {
      const anchor = state.anchorElement;
      if (!anchor) return;
      const observer = new MutationObserver(() => {
        if (!anchor.hasAttribute("data-focus-visible")) return;
        setVisible(true);
      });
      observer.observe(anchor, { attributeFilter: ["data-focus-visible"] });
      return () => observer.disconnect();
    });
  }, [store]);
  const onClickProp = props.onClick;
  const onClick = _7FULUV5Lcjs.useEvent.call(void 0, (event) => {
    onClickProp == null ? void 0 : onClickProp(event);
    if (event.defaultPrevented) return;
    store == null ? void 0 : store.setAutoFocusOnShow(true);
  });
  const onFocusProp = props.onFocus;
  const onFocus = _7FULUV5Lcjs.useEvent.call(void 0, (event) => {
    onFocusProp == null ? void 0 : onFocusProp(event);
    if (event.defaultPrevented) return;
    setVisible(true);
  });
  const { style } = _O72M6F4Zcjs.useVisuallyHidden.call(void 0, );
  if (!visible) props = {
    ...props,
    style: {
      ...style,
      ...props.style
    }
  };
  props = {
    children: /* @__PURE__ */ _jsxruntime.jsx.call(void 0, "svg", {
      display: "block",
      fill: "none",
      stroke: "currentColor",
      strokeLinecap: "round",
      strokeLinejoin: "round",
      strokeWidth: 1.5,
      viewBox: "0 0 16 16",
      height: "1em",
      width: "1em",
      children: /* @__PURE__ */ _jsxruntime.jsx.call(void 0, "polyline", { points: "4,6 8,10 12,6" })
    }),
    ...props,
    ref: _7FULUV5Lcjs.useMergeRefs.call(void 0, store.setDisclosureElement, props.ref),
    onClick,
    onFocus
  };
  props = _FWOU7QANcjs.useDialogDisclosure.call(void 0, {
    store,
    ...props
  });
  return props;
});
var HovercardDisclosure = _7FULUV5Lcjs.forwardRef.call(void 0, function HovercardDisclosure2(props) {
  return _7FULUV5Lcjs.createElement.call(void 0, TagName2, useHovercardDisclosure(props));
});

// ../ariakit-react-components/dist/hovercard/hovercard-provider.js

function HovercardProvider(props = {}) {
  return /* @__PURE__ */ _jsxruntime.jsx.call(void 0, _MU4IYYTFcjs.HovercardContextProvider, {
    value: _MU4IYYTFcjs.useHovercardStore.call(void 0, props),
    children: props.children
  });
}





exports.HovercardArrow = HovercardArrow; exports.HovercardDisclosure = HovercardDisclosure; exports.HovercardProvider = HovercardProvider;
