// @ariakit/react/cjs/__chunks/IXRLCM4X.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});








var _7ALPAKHFcjs = require('./7ALPAKHF.cjs');


var _KWYRMCLAcjs = require('./KWYRMCLA.cjs');




var _TJNZ4CWTcjs = require('./TJNZ4CWT.cjs');


var _62Q6LBYZcjs = require('./62Q6LBYZ.cjs');





















var _7FULUV5Lcjs = require('./7FULUV5L.cjs');

// ../ariakit-react-components/dist/composite/composite.js
var _react = require('react');
var _jsxruntime = require('react/jsx-runtime');
var TagName = "div";
function isGrid(items) {
  return items.some((item) => !!item.rowId);
}
function isPrintableKey(event) {
  const target = event.target;
  if (target && !_7FULUV5Lcjs.isTextField.call(void 0, target)) return false;
  return event.key.length === 1 && !event.ctrlKey && !event.metaKey;
}
function isModifierKey(event) {
  return event.key === "Shift" || event.key === "Control" || event.key === "Alt" || event.key === "Meta";
}
function useKeyboardEventProxy(store, onKeyboardEvent, previousElementRef) {
  return _7FULUV5Lcjs.useEvent.call(void 0, (event) => {
    var _a;
    onKeyboardEvent == null ? void 0 : onKeyboardEvent(event);
    if (event.defaultPrevented) return;
    if (event.isPropagationStopped()) return;
    if (!_7FULUV5Lcjs.isSelfTarget.call(void 0, event)) return;
    if (isModifierKey(event)) return;
    if (isPrintableKey(event)) return;
    const activeElement = (_a = _7ALPAKHFcjs.getEnabledItem.call(void 0, store, store.getState().activeId)) == null ? void 0 : _a.element;
    if (!activeElement) return;
    const { view, ...eventInit } = event;
    if (activeElement !== (previousElementRef == null ? void 0 : previousElementRef.current)) activeElement.focus();
    if (!_7FULUV5Lcjs.fireKeyboardEvent.call(void 0, activeElement, event.type, eventInit)) event.preventDefault();
    if (event.currentTarget.contains(activeElement)) event.stopPropagation();
  });
}
function findFirstEnabledItemInTheLastRow(items) {
  return _7ALPAKHFcjs.findFirstEnabledItem.call(void 0, _7FULUV5Lcjs.flatten2DArray.call(void 0, _7FULUV5Lcjs.reverseArray.call(void 0, _7ALPAKHFcjs.groupItemsByRows.call(void 0, items))));
}
function withBaseScrollPreserved(store, callback) {
  const { virtualFocus, baseElement } = store.getState();
  if (!virtualFocus || !baseElement || !_7FULUV5Lcjs.isTextField.call(void 0, baseElement)) {
    callback();
    return;
  }
  const savedScrollLeft = baseElement.scrollLeft;
  const savedScrollTop = baseElement.scrollTop;
  callback();
  baseElement.scrollLeft = savedScrollLeft;
  baseElement.scrollTop = savedScrollTop;
}
function useScheduleFocus(store) {
  const [scheduled, setScheduled] = _react.useState.call(void 0, false);
  const schedule = _react.useCallback.call(void 0, () => setScheduled(true), []);
  const activeItem = _TJNZ4CWTcjs.useStoreState.call(void 0, store, (state) => _7ALPAKHFcjs.getEnabledItem.call(void 0, store, state.activeId));
  _react.useEffect.call(void 0, () => {
    const activeElement = activeItem == null ? void 0 : activeItem.element;
    if (!scheduled) return;
    if (!activeElement) return;
    setScheduled(false);
    withBaseScrollPreserved(store, () => {
      activeElement.focus({ preventScroll: true });
    });
  }, [
    store,
    activeItem,
    scheduled
  ]);
  return schedule;
}
var useComposite = _7FULUV5Lcjs.createHook.call(void 0, function useComposite2({ store, composite = true, focusOnMove = composite, moveOnKeyPress = true, ...props }) {
  const context = _7ALPAKHFcjs.useCompositeProviderContext.call(void 0, );
  store = store || context;
  _7FULUV5Lcjs.invariant.call(void 0, store, process.env.NODE_ENV !== "production" && "Composite must receive a `store` prop or be wrapped in a CompositeProvider component.");
  const ref = _react.useRef.call(void 0, null);
  const previousElementRef = _react.useRef.call(void 0, null);
  const scheduleFocus = useScheduleFocus(store);
  const moves = _TJNZ4CWTcjs.useStoreState.call(void 0, store, "moves");
  const [, setBaseElement] = _7FULUV5Lcjs.useTransactionState.call(void 0, composite ? store.setBaseElement : null);
  _react.useEffect.call(void 0, () => {
    var _a;
    if (!store) return;
    if (!moves) return;
    if (!composite) return;
    if (!focusOnMove) return;
    const { activeId: activeId2 } = store.getState();
    const itemElement = (_a = _7ALPAKHFcjs.getEnabledItem.call(void 0, store, activeId2)) == null ? void 0 : _a.element;
    if (!itemElement) return;
    withBaseScrollPreserved(store, () => _7FULUV5Lcjs.focusIntoView.call(void 0, itemElement));
  }, [
    store,
    moves,
    composite,
    focusOnMove
  ]);
  _7FULUV5Lcjs.useSafeLayoutEffect.call(void 0, () => {
    if (!store) return;
    if (!moves) return;
    if (!composite) return;
    const { baseElement, activeId: activeId2 } = store.getState();
    if (!(activeId2 === null)) return;
    if (!baseElement) return;
    const previousElement = previousElementRef.current;
    previousElementRef.current = null;
    if (previousElement) _7FULUV5Lcjs.fireBlurEvent.call(void 0, previousElement, { relatedTarget: baseElement });
    if (!_7FULUV5Lcjs.hasFocus.call(void 0, baseElement)) baseElement.focus();
  }, [
    store,
    moves,
    composite
  ]);
  const activeId = _TJNZ4CWTcjs.useStoreState.call(void 0, store, "activeId");
  const virtualFocus = _TJNZ4CWTcjs.useStoreState.call(void 0, store, "virtualFocus");
  _7FULUV5Lcjs.useSafeLayoutEffect.call(void 0, () => {
    var _a;
    if (!store) return;
    if (!composite) return;
    if (!virtualFocus) return;
    const previousElement = previousElementRef.current;
    previousElementRef.current = null;
    if (!previousElement) return;
    const relatedTarget = ((_a = _7ALPAKHFcjs.getEnabledItem.call(void 0, store, activeId)) == null ? void 0 : _a.element) || _7FULUV5Lcjs.getActiveElement.call(void 0, previousElement);
    if (relatedTarget === previousElement) return;
    _7FULUV5Lcjs.fireBlurEvent.call(void 0, previousElement, { relatedTarget });
  }, [
    store,
    activeId,
    virtualFocus,
    composite
  ]);
  const onKeyDownCapture = useKeyboardEventProxy(store, props.onKeyDownCapture, previousElementRef);
  const onKeyUpCapture = useKeyboardEventProxy(store, props.onKeyUpCapture, previousElementRef);
  const onFocusCaptureProp = props.onFocusCapture;
  const onFocusCapture = _7FULUV5Lcjs.useEvent.call(void 0, (event) => {
    onFocusCaptureProp == null ? void 0 : onFocusCaptureProp(event);
    if (event.defaultPrevented) return;
    if (!store) return;
    const { virtualFocus: virtualFocus2 } = store.getState();
    if (!virtualFocus2) return;
    const previousActiveElement = event.relatedTarget;
    const isSilentlyFocused = _7ALPAKHFcjs.silentlyFocused.call(void 0, event.currentTarget);
    if (_7FULUV5Lcjs.isSelfTarget.call(void 0, event) && isSilentlyFocused) {
      event.stopPropagation();
      previousElementRef.current = previousActiveElement;
    }
  });
  const onFocusProp = props.onFocus;
  const onFocus = _7FULUV5Lcjs.useEvent.call(void 0, (event) => {
    onFocusProp == null ? void 0 : onFocusProp(event);
    if (event.defaultPrevented) return;
    if (!composite) return;
    if (!store) return;
    const { relatedTarget } = event;
    const { virtualFocus: virtualFocus2 } = store.getState();
    if (virtualFocus2) {
      if (_7FULUV5Lcjs.isSelfTarget.call(void 0, event) && !_7ALPAKHFcjs.isItem.call(void 0, store, relatedTarget)) queueMicrotask(scheduleFocus);
    } else if (_7FULUV5Lcjs.isSelfTarget.call(void 0, event)) store.setActiveId(null);
  });
  const onBlurCaptureProp = props.onBlurCapture;
  const onBlurCapture = _7FULUV5Lcjs.useEvent.call(void 0, (event) => {
    var _a;
    onBlurCaptureProp == null ? void 0 : onBlurCaptureProp(event);
    if (event.defaultPrevented) return;
    if (!store) return;
    const { virtualFocus: virtualFocus2, activeId: activeId2 } = store.getState();
    if (!virtualFocus2) return;
    const activeElement = (_a = _7ALPAKHFcjs.getEnabledItem.call(void 0, store, activeId2)) == null ? void 0 : _a.element;
    const nextActiveElement = event.relatedTarget;
    const nextActiveElementIsItem = _7ALPAKHFcjs.isItem.call(void 0, store, nextActiveElement);
    const previousElement = previousElementRef.current;
    previousElementRef.current = null;
    if (_7FULUV5Lcjs.isSelfTarget.call(void 0, event) && nextActiveElementIsItem) {
      if (nextActiveElement === activeElement) {
        if (previousElement && previousElement !== nextActiveElement) _7FULUV5Lcjs.fireBlurEvent.call(void 0, previousElement, event);
      } else if (activeElement) _7FULUV5Lcjs.fireBlurEvent.call(void 0, activeElement, event);
      else if (previousElement) _7FULUV5Lcjs.fireBlurEvent.call(void 0, previousElement, event);
      event.stopPropagation();
    } else if (!_7ALPAKHFcjs.isItem.call(void 0, store, event.target) && activeElement) _7FULUV5Lcjs.fireBlurEvent.call(void 0, activeElement, event);
  });
  const onKeyDownProp = props.onKeyDown;
  const moveOnKeyPressProp = _7FULUV5Lcjs.useBooleanEvent.call(void 0, moveOnKeyPress);
  const onKeyDown = _7FULUV5Lcjs.useEvent.call(void 0, (event) => {
    var _a, _b;
    onKeyDownProp == null ? void 0 : onKeyDownProp(event);
    if (event.nativeEvent.isComposing) return;
    if (event.defaultPrevented) return;
    if (!store) return;
    if (!_7FULUV5Lcjs.isSelfTarget.call(void 0, event)) return;
    const { orientation, renderedItems, activeId: activeId2 } = store.getState();
    if ((_b = (_a = _7ALPAKHFcjs.getEnabledItem.call(void 0, store, activeId2)) == null ? void 0 : _a.element) == null ? void 0 : _b.isConnected) return;
    const isVertical = orientation !== "horizontal";
    const isHorizontal = orientation !== "vertical";
    const grid = isGrid(renderedItems);
    if ((event.key === "ArrowLeft" || event.key === "ArrowRight" || event.key === "Home" || event.key === "End") && _7FULUV5Lcjs.isTextField.call(void 0, event.currentTarget)) return;
    const up = () => {
      var _a2;
      if (grid) return (_a2 = findFirstEnabledItemInTheLastRow(renderedItems)) == null ? void 0 : _a2.id;
      return store == null ? void 0 : store.last();
    };
    const action = {
      ArrowUp: (grid || isVertical) && up,
      ArrowRight: (grid || isHorizontal) && store.first,
      ArrowDown: (grid || isVertical) && store.first,
      ArrowLeft: (grid || isHorizontal) && store.last,
      Home: store.first,
      End: store.last,
      PageUp: store.first,
      PageDown: store.last
    }[event.key];
    if (action) {
      const id = action();
      if (id !== void 0) {
        if (!moveOnKeyPressProp(event)) return;
        event.preventDefault();
        store.move(id);
      }
    }
  });
  props = _7FULUV5Lcjs.useWrapElement.call(void 0, props, (element) => /* @__PURE__ */ _jsxruntime.jsx.call(void 0, _7ALPAKHFcjs.CompositeScopedContextProvider, {
    value: store,
    children: element
  }), [store]);
  props = {
    "aria-activedescendant": _TJNZ4CWTcjs.useStoreState.call(void 0, store, (state) => {
      var _a;
      if (!store) return;
      if (!composite) return;
      if (!state.virtualFocus) return;
      return (_a = _7ALPAKHFcjs.getEnabledItem.call(void 0, store, state.activeId)) == null ? void 0 : _a.id;
    }),
    ...props,
    ref: _7FULUV5Lcjs.useMergeRefs.call(void 0, ref, setBaseElement, props.ref),
    onKeyDownCapture,
    onKeyUpCapture,
    onFocusCapture,
    onFocus,
    onBlurCapture,
    onKeyDown
  };
  props = _62Q6LBYZcjs.useFocusable.call(void 0, {
    focusable: _TJNZ4CWTcjs.useStoreState.call(void 0, store, (state) => composite && (state.virtualFocus || state.activeId === null)),
    ...props
  });
  return props;
});
var Composite = _7FULUV5Lcjs.forwardRef.call(void 0, function Composite2(props) {
  return _7FULUV5Lcjs.createElement.call(void 0, TagName, useComposite(props));
});

// ../ariakit-react-components/dist/composite/composite-store.js
function useCompositeStoreOptions(props) {
  return {
    id: _7FULUV5Lcjs.useId.call(void 0, props.id),
    ...props
  };
}
function useCompositeStoreProps(store, update, props) {
  store = _KWYRMCLAcjs.useCollectionStoreProps.call(void 0, store, update, props);
  _TJNZ4CWTcjs.useStoreProps.call(void 0, store, props, "activeId", "setActiveId");
  _TJNZ4CWTcjs.useStoreProps.call(void 0, store, props, "includesBaseElement");
  _TJNZ4CWTcjs.useStoreProps.call(void 0, store, props, "virtualFocus");
  _TJNZ4CWTcjs.useStoreProps.call(void 0, store, props, "orientation");
  _TJNZ4CWTcjs.useStoreProps.call(void 0, store, props, "rtl");
  _TJNZ4CWTcjs.useStoreProps.call(void 0, store, props, "focusLoop");
  _TJNZ4CWTcjs.useStoreProps.call(void 0, store, props, "focusWrap");
  _TJNZ4CWTcjs.useStoreProps.call(void 0, store, props, "focusShift");
  return store;
}
function useCompositeStore(props = {}) {
  props = useCompositeStoreOptions(props);
  const [store, update] = _TJNZ4CWTcjs.useStore.call(void 0, _7ALPAKHFcjs.createCompositeStore, props);
  return useCompositeStoreProps(store, update, props);
}







exports.useComposite = useComposite; exports.Composite = Composite; exports.useCompositeStoreOptions = useCompositeStoreOptions; exports.useCompositeStoreProps = useCompositeStoreProps; exports.useCompositeStore = useCompositeStore;
