// @ariakit/react/cjs/__chunks/LSO5PXDT.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});


var _5X2KE4NBcjs = require('./5X2KE4NB.cjs');



var _7ALPAKHFcjs = require('./7ALPAKHF.cjs');


var _7FULUV5Lcjs = require('./7FULUV5L.cjs');

// ../ariakit-react-components/dist/combobox/combobox-context.js
var _react = require('react');
var ComboboxListRoleContext = _react.createContext.call(void 0, void 0);
var ctx = _7FULUV5Lcjs.createStoreContext.call(void 0, [_5X2KE4NBcjs.PopoverContextProvider, _7ALPAKHFcjs.CompositeContextProvider], [_5X2KE4NBcjs.PopoverScopedContextProvider, _7ALPAKHFcjs.CompositeScopedContextProvider]);
var useComboboxContext = ctx.useContext;
var useComboboxScopedContext = ctx.useScopedContext;
var useComboboxProviderContext = ctx.useProviderContext;
var ComboboxContextProvider = ctx.ContextProvider;
var ComboboxScopedContextProvider = ctx.ScopedContextProvider;
var ComboboxItemValueContext = _react.createContext.call(void 0, void 0);
var ComboboxItemCheckedContext = _react.createContext.call(void 0, false);










exports.ComboboxListRoleContext = ComboboxListRoleContext; exports.useComboboxContext = useComboboxContext; exports.useComboboxScopedContext = useComboboxScopedContext; exports.useComboboxProviderContext = useComboboxProviderContext; exports.ComboboxContextProvider = ComboboxContextProvider; exports.ComboboxScopedContextProvider = ComboboxScopedContextProvider; exports.ComboboxItemValueContext = ComboboxItemValueContext; exports.ComboboxItemCheckedContext = ComboboxItemCheckedContext;
