// @ariakit/react/cjs/__chunks/MO56HI6H.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});

var _KWYRMCLAcjs = require('./KWYRMCLA.cjs');








var _7FULUV5Lcjs = require('./7FULUV5L.cjs');

// ../ariakit-react-components/dist/collection/collection-item.js
var _react = require('react');
var TagName = "div";
var useCollectionItem = _7FULUV5Lcjs.createHook.call(void 0, function useCollectionItem2({ store, shouldRegisterItem = true, getItem = _7FULUV5Lcjs.identity, element, ...props }) {
  const context = _KWYRMCLAcjs.useCollectionContext.call(void 0, );
  store = store || context;
  const id = _7FULUV5Lcjs.useId.call(void 0, props.id);
  const ref = _react.useRef.call(void 0, element);
  _react.useEffect.call(void 0, () => {
    const element2 = ref.current;
    if (!id) return;
    if (!element2) return;
    if (!shouldRegisterItem) return;
    const item = getItem({
      id,
      element: element2
    });
    return store == null ? void 0 : store.renderItem(item);
  }, [
    id,
    shouldRegisterItem,
    getItem,
    store
  ]);
  props = {
    ...props,
    ref: _7FULUV5Lcjs.useMergeRefs.call(void 0, ref, props.ref)
  };
  return _7FULUV5Lcjs.removeUndefinedValues.call(void 0, props);
});
var CollectionItem = _7FULUV5Lcjs.forwardRef.call(void 0, function CollectionItem2(props) {
  return _7FULUV5Lcjs.createElement.call(void 0, TagName, useCollectionItem(props));
});




exports.useCollectionItem = useCollectionItem; exports.CollectionItem = CollectionItem;
