// @ariakit/react/cjs/__chunks/RB3KDI27.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});

var _7ALPAKHFcjs = require('./7ALPAKHF.cjs');



var _F5C4EN5Bcjs = require('./F5C4EN5B.cjs');
















var _7FULUV5Lcjs = require('./7FULUV5L.cjs');

// ../ariakit-react-components/dist/composite/composite-group.js
var TagName = "div";
var useCompositeGroup = _7FULUV5Lcjs.createHook.call(void 0, function useCompositeGroup2({ store, ...props }) {
  props = _F5C4EN5Bcjs.useGroup.call(void 0, props);
  return props;
});
var CompositeGroup = _7FULUV5Lcjs.forwardRef.call(void 0, function CompositeGroup2(props) {
  return _7FULUV5Lcjs.createElement.call(void 0, TagName, useCompositeGroup(props));
});

// ../ariakit-react-components/dist/composite/composite-group-label.js
var TagName2 = "div";
var useCompositeGroupLabel = _7FULUV5Lcjs.createHook.call(void 0, function useCompositeGroupLabel2({ store, ...props }) {
  props = _F5C4EN5Bcjs.useGroupLabel.call(void 0, props);
  return props;
});
var CompositeGroupLabel = _7FULUV5Lcjs.forwardRef.call(void 0, function CompositeGroupLabel2(props) {
  return _7FULUV5Lcjs.createElement.call(void 0, TagName2, useCompositeGroupLabel(props));
});

// ../ariakit-react-components/dist/composite/composite-hover.js
var _react = require('react');
var TagName3 = "div";
function hoveringInside(event) {
  const nextElement = event.relatedTarget;
  if (!_7FULUV5Lcjs.isElement.call(void 0, nextElement)) return false;
  return _7FULUV5Lcjs.contains.call(void 0, event.currentTarget, nextElement);
}
var symbol = /* @__PURE__ */ Symbol("composite-hover");
function movingToAnotherItem(event) {
  const { relatedTarget } = event;
  if (!_7FULUV5Lcjs.isElement.call(void 0, relatedTarget)) return false;
  let dest = relatedTarget;
  do {
    if (_7FULUV5Lcjs.hasOwnProperty.call(void 0, dest, symbol) && dest[symbol]) return true;
    dest = dest.parentElement;
  } while (dest);
  return false;
}
var useCompositeHover = _7FULUV5Lcjs.createHook.call(void 0, function useCompositeHover2({ store, focusOnHover = true, blurOnHoverEnd = !!focusOnHover, ...props }) {
  const context = _7ALPAKHFcjs.useCompositeScopedContext.call(void 0, );
  store = store || context;
  _7FULUV5Lcjs.invariant.call(void 0, store, process.env.NODE_ENV !== "production" && "CompositeHover must be wrapped in a Composite component.");
  const isMouseMoving = _7FULUV5Lcjs.useIsMouseMoving.call(void 0, );
  const onMouseMoveProp = props.onMouseMove;
  const focusOnHoverProp = _7FULUV5Lcjs.useBooleanEvent.call(void 0, focusOnHover);
  const onMouseMove = _7FULUV5Lcjs.useEvent.call(void 0, (event) => {
    onMouseMoveProp == null ? void 0 : onMouseMoveProp(event);
    if (event.defaultPrevented) return;
    if (!isMouseMoving()) return;
    if (!focusOnHoverProp(event)) return;
    if (!_7FULUV5Lcjs.hasFocusWithin.call(void 0, event.currentTarget)) {
      const baseElement = store == null ? void 0 : store.getState().baseElement;
      if (baseElement && !_7FULUV5Lcjs.hasFocus.call(void 0, baseElement)) baseElement.focus();
    }
    store == null ? void 0 : store.setActiveId(event.currentTarget.id);
  });
  const onMouseLeaveProp = props.onMouseLeave;
  const blurOnHoverEndProp = _7FULUV5Lcjs.useBooleanEvent.call(void 0, blurOnHoverEnd);
  const onMouseLeave = _7FULUV5Lcjs.useEvent.call(void 0, (event) => {
    var _a;
    onMouseLeaveProp == null ? void 0 : onMouseLeaveProp(event);
    if (event.defaultPrevented) return;
    if (!isMouseMoving()) return;
    if (hoveringInside(event)) return;
    if (movingToAnotherItem(event)) return;
    if (!focusOnHoverProp(event)) return;
    if (!blurOnHoverEndProp(event)) return;
    store == null ? void 0 : store.setActiveId(null);
    (_a = store == null ? void 0 : store.getState().baseElement) == null ? void 0 : _a.focus();
  });
  const ref = _react.useCallback.call(void 0, (element) => {
    if (!element) return;
    element[symbol] = true;
  }, []);
  props = {
    ...props,
    ref: _7FULUV5Lcjs.useMergeRefs.call(void 0, ref, props.ref),
    onMouseMove,
    onMouseLeave
  };
  return _7FULUV5Lcjs.removeUndefinedValues.call(void 0, props);
});
var CompositeHover = _7FULUV5Lcjs.memo.call(void 0, _7FULUV5Lcjs.forwardRef.call(void 0, function CompositeHover2(props) {
  return _7FULUV5Lcjs.createElement.call(void 0, TagName3, useCompositeHover(props));
}));








exports.useCompositeGroup = useCompositeGroup; exports.CompositeGroup = CompositeGroup; exports.useCompositeGroupLabel = useCompositeGroupLabel; exports.CompositeGroupLabel = CompositeGroupLabel; exports.useCompositeHover = useCompositeHover; exports.CompositeHover = CompositeHover;
