// @ariakit/react/cjs/__chunks/T4LX3AM2.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});



var _SICDNNB6cjs = require('./SICDNNB6.cjs');




var _IXRLCM4Xcjs = require('./IXRLCM4X.cjs');


var _7ALPAKHFcjs = require('./7ALPAKHF.cjs');




var _TJNZ4CWTcjs = require('./TJNZ4CWT.cjs');







var _7FULUV5Lcjs = require('./7FULUV5L.cjs');

// ../ariakit-react-components/dist/radio/radio-group.js
var _jsxruntime = require('react/jsx-runtime');
var TagName = "div";
var useRadioGroup = _7FULUV5Lcjs.createHook.call(void 0, function useRadioGroup2({ store, ...props }) {
  const context = _SICDNNB6cjs.useRadioProviderContext.call(void 0, );
  store = store || context;
  _7FULUV5Lcjs.invariant.call(void 0, store, process.env.NODE_ENV !== "production" && "RadioGroup must receive a `store` prop or be wrapped in a RadioProvider component.");
  props = _7FULUV5Lcjs.useWrapElement.call(void 0, props, (element) => /* @__PURE__ */ _jsxruntime.jsx.call(void 0, _SICDNNB6cjs.RadioScopedContextProvider, {
    value: store,
    children: element
  }), [store]);
  props = {
    role: "radiogroup",
    ...props
  };
  props = _IXRLCM4Xcjs.useComposite.call(void 0, {
    store,
    ...props
  });
  return props;
});
var RadioGroup = _7FULUV5Lcjs.forwardRef.call(void 0, function RadioGroup2(props) {
  return _7FULUV5Lcjs.createElement.call(void 0, TagName, useRadioGroup(props));
});

// ../ariakit-components/dist/radio/radio-store.js
function createRadioStore(props = {}) {
  var _a;
  const syncState = (_a = props.store) == null ? void 0 : _a.getState();
  const composite = _7ALPAKHFcjs.createCompositeStore.call(void 0, {
    ...props,
    focusLoop: _7FULUV5Lcjs.defaultValue.call(void 0, props.focusLoop, syncState == null ? void 0 : syncState.focusLoop, true)
  });
  const radio = _TJNZ4CWTcjs.createStore.call(void 0, {
    ...composite.getState(),
    value: _7FULUV5Lcjs.defaultValue.call(void 0, props.value, syncState == null ? void 0 : syncState.value, props.defaultValue, null)
  }, composite, props.store);
  return {
    ...composite,
    ...radio,
    setValue: (value) => radio.setState("value", value)
  };
}

// ../ariakit-react-components/dist/radio/radio-store.js
function useRadioStoreProps(store, update, props) {
  store = _IXRLCM4Xcjs.useCompositeStoreProps.call(void 0, store, update, props);
  _TJNZ4CWTcjs.useStoreProps.call(void 0, store, props, "value", "setValue");
  return store;
}
function useRadioStore(props = {}) {
  props = _IXRLCM4Xcjs.useCompositeStoreOptions.call(void 0, props);
  const [store, update] = _TJNZ4CWTcjs.useStore.call(void 0, createRadioStore, props);
  return useRadioStoreProps(store, update, props);
}

// ../ariakit-react-components/dist/radio/radio-provider.js

function RadioProvider(props = {}) {
  return /* @__PURE__ */ _jsxruntime.jsx.call(void 0, _SICDNNB6cjs.RadioContextProvider, {
    value: useRadioStore(props),
    children: props.children
  });
}





exports.RadioGroup = RadioGroup; exports.useRadioStore = useRadioStore; exports.RadioProvider = RadioProvider;
