// @ariakit/react/cjs/__chunks/UYTV7GOW.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});


var _IXRLCM4Xcjs = require('./IXRLCM4X.cjs');




var _7ALPAKHFcjs = require('./7ALPAKHF.cjs');




var _TJNZ4CWTcjs = require('./TJNZ4CWT.cjs');







var _7FULUV5Lcjs = require('./7FULUV5L.cjs');

// ../ariakit-react-components/dist/menubar/menubar-context.js
var menubar = _7FULUV5Lcjs.createStoreContext.call(void 0, [_7ALPAKHFcjs.CompositeContextProvider], [_7ALPAKHFcjs.CompositeScopedContextProvider]);
var useMenubarContext = menubar.useContext;
var useMenubarScopedContext = menubar.useScopedContext;
var useMenubarProviderContext = menubar.useProviderContext;
var MenubarContextProvider = menubar.ContextProvider;
var MenubarScopedContextProvider = menubar.ScopedContextProvider;

// ../ariakit-components/dist/menubar/menubar-store.js
function createMenubarStore(props = {}) {
  var _a;
  const syncState = (_a = props.store) == null ? void 0 : _a.getState();
  const composite = _7ALPAKHFcjs.createCompositeStore.call(void 0, {
    ...props,
    orientation: _7FULUV5Lcjs.defaultValue.call(void 0, props.orientation, syncState == null ? void 0 : syncState.orientation, "horizontal"),
    focusLoop: _7FULUV5Lcjs.defaultValue.call(void 0, props.focusLoop, syncState == null ? void 0 : syncState.focusLoop, true)
  });
  const menubar2 = _TJNZ4CWTcjs.createStore.call(void 0, { ...composite.getState() }, composite, props.store);
  return {
    ...composite,
    ...menubar2
  };
}

// ../ariakit-react-components/dist/menubar/menubar-store.js
function useMenubarStoreProps(store, update, props) {
  return _IXRLCM4Xcjs.useCompositeStoreProps.call(void 0, store, update, props);
}
function useMenubarStore(props = {}) {
  const [store, update] = _TJNZ4CWTcjs.useStore.call(void 0, createMenubarStore, props);
  return useMenubarStoreProps(store, update, props);
}

// ../ariakit-react-components/dist/menubar/menubar.js
var _jsxruntime = require('react/jsx-runtime');
var TagName = "div";
var useMenubar = _7FULUV5Lcjs.createHook.call(void 0, function useMenubar2({ store: storeProp, composite = true, orientation: orientationProp, virtualFocus, focusLoop, rtl, ...props }) {
  const context = useMenubarProviderContext();
  storeProp = storeProp || context;
  const store = useMenubarStore({
    store: storeProp,
    orientation: orientationProp,
    virtualFocus,
    focusLoop,
    rtl
  });
  const orientation = _TJNZ4CWTcjs.useStoreState.call(void 0, store, (state) => !composite || state.orientation === "both" ? void 0 : state.orientation);
  props = _7FULUV5Lcjs.useWrapElement.call(void 0, props, (element) => /* @__PURE__ */ _jsxruntime.jsx.call(void 0, MenubarScopedContextProvider, {
    value: store,
    children: element
  }), [store]);
  if (composite) props = {
    role: "menubar",
    "aria-orientation": orientation,
    ...props
  };
  props = _IXRLCM4Xcjs.useComposite.call(void 0, {
    store,
    composite,
    ...props
  });
  return props;
});
var Menubar = _7FULUV5Lcjs.forwardRef.call(void 0, function Menubar2(props) {
  return _7FULUV5Lcjs.createElement.call(void 0, TagName, useMenubar(props));
});

// ../ariakit-react-components/dist/menubar/menubar-provider.js

function MenubarProvider(props = {}) {
  return /* @__PURE__ */ _jsxruntime.jsx.call(void 0, MenubarContextProvider, {
    value: useMenubarStore(props),
    children: props.children
  });
}








exports.useMenubarContext = useMenubarContext; exports.useMenubarScopedContext = useMenubarScopedContext; exports.useMenubarStore = useMenubarStore; exports.useMenubar = useMenubar; exports.Menubar = Menubar; exports.MenubarProvider = MenubarProvider;
