// @ariakit/react/cjs/__chunks/ZNYKFHPV.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});


var _5X2KE4NBcjs = require('./5X2KE4NB.cjs');



var _7ALPAKHFcjs = require('./7ALPAKHF.cjs');


var _7FULUV5Lcjs = require('./7FULUV5L.cjs');

// ../ariakit-react-components/dist/select/select-context.js
var _react = require('react');
var ctx = _7FULUV5Lcjs.createStoreContext.call(void 0, [_5X2KE4NBcjs.PopoverContextProvider, _7ALPAKHFcjs.CompositeContextProvider], [_5X2KE4NBcjs.PopoverScopedContextProvider, _7ALPAKHFcjs.CompositeScopedContextProvider]);
var useSelectContext = ctx.useContext;
var useSelectScopedContext = ctx.useScopedContext;
var useSelectProviderContext = ctx.useProviderContext;
var SelectContextProvider = ctx.ContextProvider;
var SelectScopedContextProvider = ctx.ScopedContextProvider;
var SelectItemCheckedContext = _react.createContext.call(void 0, false);
var SelectHeadingContext = _react.createContext.call(void 0, null);









exports.useSelectContext = useSelectContext; exports.useSelectScopedContext = useSelectScopedContext; exports.useSelectProviderContext = useSelectProviderContext; exports.SelectContextProvider = SelectContextProvider; exports.SelectScopedContextProvider = SelectScopedContextProvider; exports.SelectItemCheckedContext = SelectItemCheckedContext; exports.SelectHeadingContext = SelectHeadingContext;
